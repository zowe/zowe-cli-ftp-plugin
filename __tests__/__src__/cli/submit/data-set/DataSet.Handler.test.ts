/*
 * This program and the accompanying materials are made available under the terms of the
 * Eclipse Public License v2.0 which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-v20.html
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Copyright Contributors to the Zowe Project.
 *
 */

import SubmitJobFromLocalFileHandler from "../../../../../src/cli/submit/data-set/DataSet.Handler";
import TestUtils from "../../TestUtils";

describe("Submit data set handler", () => {

    it("should return correct message if the data set is submit successfully.", async () => {
        const handler = new SubmitJobFromLocalFileHandler();
        const jobDetails = {
            jobname: "jobname1",
            jobid: "jobid1"
        };

        const mockResponse = TestUtils.getMockResponse();
        const mockParams: any = {
            arguments: {
                dataSet: "ds1"
            },
            connection: {
                getDataset: jest.fn().mockReturnValue(Promise.resolve(Buffer.from(""))),
                submitJCL: jest.fn().mockReturnValue(Promise.resolve("jobid1")),
                getJobStatus: jest.fn().mockReturnValue(Promise.resolve(jobDetails))
            },
            response: mockResponse
        };
        await handler.processFTP(mockParams);
        expect(mockResponse.data.setObj.mock.calls[0][0]).toMatchSnapshot();
        expect(mockResponse.format.output.mock.calls[0][0]).toMatchSnapshot();
    });

});
