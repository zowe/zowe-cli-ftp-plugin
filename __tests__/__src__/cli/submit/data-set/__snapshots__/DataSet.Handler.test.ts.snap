// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Submit data set handler should return correct message if the data set is submit successfully. 1`] = `
Object {
  "jobid": "jobid1",
  "jobname": "jobname1",
}
`;

exports[`Submit data set handler should return correct message if the data set is submit successfully. 2`] = `
Object {
  "fields": Array [
    "jobid",
    "jobname",
    "owner",
    "status",
  ],
  "format": "object",
  "output": Object {
    "jobid": "jobid1",
    "jobname": "jobname1",
  },
}
`;
